backend/domain/Post.ts

```diff
@@ -0,0 +1,76 @@
+import { RowDataPacket } from "mysql2";
+
+export interface IPost extends RowDataPacket {
+  id: number;
+  title: string;
+  locations: string;
+  price: number;
+  state: string;
+  photos: string;
+  view: number;
+  likeCount?: number;
+  roomCount?: number;
+  categoryId: number;
+  categoryName?: string;
+  userId: number;
+  username?: string;
+  createdAt: string;
+  updatedAt?: string;
+}
+
+class Post {
+  id: number;
+  title: string;
+  location: string;
+  price: number;
+  state: string;
+  photos: string[];
+  view: number;
+  likeCount?: number;
+  roomCount?: number;
+  categoryId: number;
+  categoryName?: string;
+  userId: number;
+  username?: string;
+  createdAt: Date;
+  updatedAt?: Date;
+
+  constructor(post: IPost) {
+    const {
+      id,
+      title,
+      location,
+      price,
+      state,
+      photos,
+      view,
+      likeCount,
+      roomCount,
+      categoryId,
+      categoryName,
+      userId,
+      username,
+      createdAt,
+      updatedAt,
+    } = post;
+    this.id = id;
+    this.title = title;
+    this.location = location;
+    this.price = price;
+    this.state = state;
+    this.photos = photos.split(".").map((photo) => photo.trim());
+    this.view = view;
+    this.likeCount = likeCount;
+    this.roomCount = roomCount;
+    this.categoryId = categoryId;
+    this.categoryName = categoryName;
```

Comment:
```suggestion
    if (categoryName) {
      this.category = new Category({ id: categoryId, name: categoryName });
    }
```

이런 식으로 Post 안에서 객체를 가질 수 있길 바랬는데, RowDataPacket 때문에 안되는군요..😥
타입스크립트...어렵네요....

이 상태라면 결국 객체 끼리의 소통은 불가능하겠네요ㅠㅠ
더 좋은 방법이 있는지 이것도 생각해볼 지점이 많은 것 같습니다.